src/frequenz/client/reporting/_client.py

```diff
@@ -160,7 +160,7 @@ async def list_single_component_data(
         metrics: Metric | list[Metric],
         start_dt: datetime,
         end_dt: datetime,
-        resolution: int | None,
+        resampling_period: timedelta,
```

Comment:
This can also be `None` if no resampling is desired.

Reply:
I get a mypy error when I set it to `None` and the use `total_seconds()` in the streaming filter. Not quite sure how to work around it?

Reply:
> _client.py:303: error: Item "None" of "timedelta | None" has no attribute "total_seconds"  [union-attr]

Reply:
If it's None you can pass that to the API, otherwise you convert it. E.g.
```
resolution = td.total_seconds() if sp is not None else None
```

Reply:
I have added this in.